### PR TITLE
Fix: Validation Rule VR.507-1 `ChargeTypeTariffPriceCountRule` to NOT block master data only requests and not throw when `Resolution = P1M`

### PR DESCRIPTION
**Pull request by prtandrup**
## Description

This PR removes some shortcomings in the way VR.507-1 `ChargeTypeTariffPriceCountRule ` is currently implemented:
* It allows master data only requests through (requests with no prices)
* It doesn't throw when resolution = P1M (monthly)

But be aware that it still rejects price series longer than a day of values.

Nevertheless, I do believe we need to reconsider this validation rule; 
* is it fit for a CIM context in its current state? 
* Should it be applied on all types of charges?
* Should it only assert that the number of Points match the time interval of the price series?
* Should we perhaps add a new validation rule for restricting price series shorter than 1 day of points?

I'll note this on the story as well for further discussion. For now, this PR will bring the validation rule to a better, but not final, place. 

## References

* #1072

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs [hunk 1]
```diff
@@ -53,7 +61,6 @@ public class ChargeTypeTariffPriceCountRuleTests
         }
 
         [Theory]
-        [InlineAutoMoqData(0, false)]
```
Comment (jonasdmoeller) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs:56]: Why remove this test for count = 0? Shouldn't it just return `true` after this update?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs [hunk 2]
```diff
@@ -77,14 +84,34 @@ public class ChargeTypeTariffPriceCountRuleTests
...
```
Comment (jonasdmoeller) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs:80]: Why remove this test for count = 0? Shouldn't it just return `true` after this update?

**Review by jonasdmoeller — DISMISSED** [review]

**Review by x-platformcoder — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs (hunk 1 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs:56]: That scenario is moved to the new `IsValid_WhenPointsCountIsZero_IsTrue()` test method, which is fine, I think, as it represents the new case that master data only requests are allowed.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs (hunk 2 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs:80]: Yes, it returns `true`, see my comment above.

**Pull request merged**